basic-algorithms.js

```diff
@@ -1,7 +1,41 @@
-// Names and Input
+//variables 
+var hacker1 = "Samuel"
+var hacker2 = window.prompt("what is the navigators name?")
 
 
-//Conditionals
+//console logs
+console.log(`The drivers name is ${hacker1}`);
+console.log(`The navigators name is ${hacker2)}`)
 
 
-// Lorem ipsum generator
+//Let's get looping
+for (var i = 0; i < hacker1.length || hacker2.length; i++){
+    if (hacker1.length > hacker2.length){
+        console.log(`The Driver has the longest name, it has ${hacker1.length} characters`)
+    } else if (hacker1.length < hacker2.length){
+        console.log(`Yo, navigator got the longest name, it has ${hacker2.length} characters`)
+    } else {}
+    console.log(`wow, you both got equally long names, ${hacker2.length} characters!!`)
+}
+
+//loop drivers name 
+  for(let i = 0; i < hacker1.length; i++){
+    console.log(`${hacker1[i]}`)
+  }
+
+  //loops driver name backwards
+  for(let i = hacker.length; i > 0; i--){
```

Comment:
here for 'samuel' backwards you'd get 'amuel', missing the first letter. To reach the first letter you need to iterate backwards and set the condition of the for loop (the middle statement) to also equal 0, just like this:

**for(let i = hacker.length; i >= 0; i--)**  You only forgot to add the equal symbol